packages/fast-components-react-base/src/typography/typography.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import Foundation, { HandledProps } from "../foundation";
+import {
+    ITypographyHandledProps,
+    ITypographyManagedClasses,
+    ITypographyUnhandledProps,
+    TypeLevel,
+    TypographyTag
+} from "./typography.props";
+import { IManagedClasses, ITypographyClassNameContract } from "@microsoft/fast-components-class-name-contracts";
+
+/* tslint:disable-next-line */
+class Typography extends Foundation<ITypographyHandledProps & IManagedClasses<ITypographyClassNameContract>, ITypographyUnhandledProps, {}> {
+    public static defaultProps: Partial<ITypographyHandledProps> = {
+        tag: TypographyTag.p
+    };
+
+    protected handledProps: HandledProps<ITypographyHandledProps & IManagedClasses<ITypographyClassNameContract>> = {
+        managedClasses: void 0,
+        tag: void 0,
+        typeLevel: void 0
+    };
+
+    /**
+     * Stores HTML tag for use in render
+     */
+    private get tag(): string {
+        return this.generateHTMLTag();
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLHeadingElement | HTMLParagraphElement | HTMLSpanElement | HTMLTableCaptionElement> {
+        return (
+            <this.tag
+                {...this.unhandledProps()}
+                className={this.generateClassNames()}
+            >
+                {this.props.children}
+            </this.tag>
+        );
+    }
+
+    protected generateClassNames(): string {
+        const classNames: string = this.getManagedClassName();
+
+        return super.generateClassNames(classNames);
+    }
+
+    /**
+     * Creates tags for rendering based on href
+     */
+    private generateHTMLTag(): string {
+        switch (this.props.tag) {
+            case TypographyTag.caption:
+                return "caption";
+            case TypographyTag.h1:
+                return "h1";
+            case TypographyTag.h2:
+                return "h2";
+            case TypographyTag.h3:
+                return "h3";
+            case TypographyTag.h4:
+                return "h4";
+            case TypographyTag.h5:
+                return "h5";
+            case TypographyTag.h6:
+                return "h6";
+            case TypographyTag.span:
+                return "span";
+            case TypographyTag.p:
+            default:
+                return "p";
+        }
+    }
+
+    private getManagedClassName(): string {
+        switch (this.props.typeLevel) {
+            case TypeLevel._1:
+            default:
```

Comment:
It might be more readable to put default above TypeLevel._1 because then it reads that the default falls through to 1